openpgp/packet/signature.go

```diff
@@ -550,6 +560,11 @@ func (sig *Signature) Sign(h hash.Hash, priv *PrivateKey, config *Config) (err e
 		return
 	}
 
+	if hashIsSigner {
+		err = signer.Sign(sig)
```

Comment:
I added a test in 21bb752d661b05f3c87a1a302b920b902b517ce0, checking that errors returned from `Signer.Sign` will be bubbled up through `Signature.Sign` and `write.SignWithSigner`.